http/src/main/java/org/sonarsource/sonarlint/core/http/ApacheHttpClientAdapter.java

```diff
@@ -265,11 +268,14 @@ public void cancel() {
   public WebSocket createWebSocketConnection(String url, Consumer<String> messageConsumer, Runnable onClosedRunnable) {
     // TODO handle handshake or other errors
     return java.net.http.HttpClient
-        .newHttpClient()
-        .newWebSocketBuilder()
-        .header("Authorization", "Bearer " + usernameOrToken)
-        .buildAsync(URI.create(url), new WebSocketClient(messageConsumer, onClosedRunnable))
-        .join();
+      .newBuilder()
+      // Don't use the default thread pool as it won't allow inheriting thread local variables
+      .executor(Executors.newCachedThreadPool(threadWithNamePrefix("sonarcloud-websocket-")))
```

Comment:
Usually, we try to shutdown created executors, isn't it needed? In fact the same question goes for the client, is there some kind of cleanup to do at the end? I might have missed that when we implemented web sockets

Reply:
I think closing the Websocket should be enough to release all resources.

Reply:
In the end I made more refactoring and I decided to close the executor service